integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```diff
@@ -748,4 +754,8 @@ class CarbonScanRDD[T: ClassTag](
     vectorReader = boolean
   }
 
+  // TODO find the better way set it.
+  def setDirectScanSupport(boolean: Boolean): Unit = {
```

Comment:
can make directScan as a constructor parameter like '''var  directScan: Boolean = false'''

Other flows can set using constructor and this method would not be required. scanRdd.directScan = true would do the job.

Reply:
Actually directScan can be passed through constructor like var directScan: Boolean = false. 

Current flows would not be impacted. 
Can be enabled using scanRDD.directScan = true

Reply:
Actually directScan can be passed through constructor like var directScan: Boolean = false. 

Current flows would not be impacted. 
Can be enabled using scanRDD.directScan = true

Reply:
Actually directScan can be passed through constructor like var directScan: Boolean = false. 

Current flows would not be impacted. 
Can be enabled using scanRDD.directScan = true

Reply:
Actually directScan can be passed through constructor like var directScan: Boolean = false. 

Current flows would not be impacted. 
Can be enabled using scanRDD.directScan = true

Reply:
Actually directScan can be passed through constructor like var directScan: Boolean = false. 

Current flows would not be impacted. 
Can be enabled using scanRDD.directScan = true

Reply:
Actually directScan can be passed through constructor like var directScan: Boolean = false. 

Current flows would not be impacted. 
Can be enabled using scanRDD.directScan = true

Reply:
Actually directScan can be passed through constructor like var directScan: Boolean = false. 

Current flows would not be impacted. 
Can be enabled using scanRDD.directScan = true

Reply:
Actually directScan can be passed through constructor like ```var directScan: Boolean = false```.
Current flows would not be impacted and can be enabled using scanRDD.directScan = true instead of this method.

Reply:
Actually directScan can be passed through constructor like ```var directScan: Boolean = false```.
Current flows would not be impacted and can be enabled using scanRDD.directScan = true instead of this method.

Reply:
Actually directScan can be passed through constructor like ```var directScan: Boolean = false```.
Current flows would not be impacted and can be enabled using scanRDD.directScan = true instead of this method.

Reply:
It is  a private variable